src/Form/EntityEmbedDialog.php

```diff
@@ -571,4 +733,32 @@ public function getDisplayPluginOptions(EmbedButtonInterface $embed_button, Enti
     natsort($plugins);
     return $plugins;
   }
+
+  /**
+   * Registers JS callback that gets entities from entity browser and updates
+   * form values accordingly.
+   */
+  public function registerJSCallback(RegisterJSCallbacks $event) {
+    if ($event->getBrowserID() == $this->entityBrowser->id()) {
+      $event->registerCallback('Drupal.entityBrowser.selectionCompleted');
+      $event->registerCallback('Drupal.entityEmbedDialog.selectionCompleted');
+    }
+  }
+
+  /**
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   */
+  protected function setEntityBrowserFormState(FormStateInterface $form_state) {
```

Comment:
Nothing is saved in form state in this function, which makes it's name a bit confusing. Maybe something like "loadEntityBrowser" or something similar?